python/recipe.cpp

```diff
@@ -110,9 +110,31 @@ std::vector<arb::event_generator> py_recipe_shim::event_generators(arb::cell_gid
     return gens;
 }
 
+struct cell_connection_shim {
+    arb::cell_member_type source;
+    arb::cell_member_type dest;
+    float weight;
+    float delay;
+
+    cell_connection_shim(arb::cell_member_type s, arb::cell_member_type d, float w, float t) {
+        source = s;
+        dest   = d;
+        weight = w;
+        set_delay(t);
+    }
+
+    // getter and setter (in order to assert when being set)
+    void set_delay(float t) {
+        pyarb::assert_throw([](auto&& f){ return f>0.; }(t), "delay must be positive");
```

Comment:
We know that the type of `delay` is `float`, so it is clearer to use `[](float f)...` instead of `[](auto&& f)...`.